src/migration/1689851339253-PromoteDateTime.ts

```diff
@@ -0,0 +1,21 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class PromoteDateTime1689851339253 implements MigrationInterface {
+  name = 'PromoteDateTime1689851339253';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`DROP INDEX "IDX_post_flags_promoteToPublic"`);
+    await queryRunner.query(
+      `CREATE INDEX "IDX_post_flags_promoteToPublic" ON post USING BTREE (((flags->>'promoteToPublic')::integer))`,
```

Comment:
Note we have to cast with `-->` operator. You can read more on it [here](https://www.postgresql.org/docs/9.5/functions-json.html#FUNCTIONS-JSON-OP-TABLE) and [here](https://stackoverflow.com/a/39173083).

Also subnote, the type can only be `integer` which doesn't allow `bigint` so we can't store milliseconds, hence I store it as seconds value.
For this specific case it should be ok, but good to note.

Reply:
Thanks for mentioning. We have to make sure platform team reads it properly